### PR TITLE
Modify  interface to also contain the  interface from the envoy contr…

### PR DESCRIPTION
**Pull request by AndrewSC208**
Modify `chassis.Logger` interface to embed the `Logger` interface from the envoy control plane go module and add a way for the application service to use the `grpc.Server` which is also needed for the envoy control plane module. The changes are necessary to integrate `fuse` and `envoy`.

### CODE REVIEW

File: pkg/chassis/rpc.go [hunk 1]
```diff
@@ -11,6 +15,13 @@ const (
 	Grpc
 )
 
+const (
+	grpcKeepaliveTime        = 30 * time.Second
```
Comment (jgkawell) [inline pkg/chassis/rpc.go:19]: Just a note that these will need to be configurable in the future

Comment (AndrewSC208) [reply pkg/chassis/rpc.go:19]: Totally agree!

Comment (AndrewSC208) [reply pkg/chassis/rpc.go:19]: Should we make backlog tickets for this sort of thing?

File: pkg/loggers/logrus/logrus.go [hunk 2]
```diff
@@ -92,6 +92,25 @@ func (l *logger) WithFields(fields chassis.Fields) chassis.Logger {
 	return newLogger(l.entry.WithFields(logrusFields))
 }
 
+// Implement `log.Logger` for `envoyproxy/go-control-plane/pkg/cache/v3`
+// go-control-plane has it's own logger interface that needs to be implemented for logging
+// to work correctly
+func (l *logger) Debugf(msg string, args ...any) {
```
Comment (jgkawell) [inline pkg/loggers/logrus/logrus.go:98]: Probably should note in the doc comment on each function that you're dropping the `args` on the floor. Or better yet add them as fields before calling the subsequent write function. You should be able to call `WithFields()` to add them.

File: pkg/chassis/rpc.go (hunk 1 above)
Comment (jgkawell) [reply pkg/chassis/rpc.go:19]: That's a good idea

File: pkg/loggers/logrus/logrus.go (hunk 2 above)
Comment (AndrewSC208) [reply pkg/loggers/logrus/logrus.go:98]: It's def not clean. :(

Addin `WithFields()` is a good idea.

Comment (AndrewSC208) [reply pkg/loggers/logrus/logrus.go:98]: Looks like `WithFields` is incompatible. It's expecting a `map[string]interface{}` and args is an `any`. I'm gonna add a comment to move on.

**Pull request merged**